python/rmm/rmm.py

```diff
@@ -105,29 +59,33 @@ def reinitialize(
         If True, enable run-time logging of all memory events
         (alloc, free, realloc).
         This has significant performance impact.
-    """
-    _finalize()
-    return _initialize(
+    log_file_name : str
+        Name of the log file. If not specified, the environment variable
+        RMM_LOG_FILE is used. A TypeError is thrown if neither is available.
+    """
+    if logging and not log_file_name:
+        log_file_name = os.getenv("RMM_LOG_FILE")
+        if not log_file_name:
+            raise TypeError(
+                "RMM log file must be specified either using "
+                "log_file_name= argument or RMM_LOG_FILE "
+                "environment variable"
+            )
```

Comment:
Any chance this could accept a file object *or* a file name?  Passing a file object lets me effectively implement a callback, which would be much nicer for my usecase than creating/parsing/cleaning up a file. If not, I can deal with a file too, but I'd rather not.

Reply:
On second thought, as much as I'd like to use a file obj, this request is probably non-trivial and maybe out-of-scope for this PR. I'll leave it unresolved here in case you want to comment.
There are examples of passing python callbacks to C extensions (and cython might make it easier), but again, probably out-of-scope here (although I'll gladly take it if you write it!)

Reply:
It's possible with some support from `librmm`. Something very similar was done in https://github.com/rapidsai/cudf/pull/5061 - @kkraus14 perhaps you could weigh in here?

Reply:
I don't think logging to a file object is in scope for a logging interface like this, but specifying a directory to write the log file to definitely sounds in scope.

Reply:
The logging adaptor can write to any arbitrary `std::ostream`. If a Python file object can be created to satisfy that interface, then it can work fine.